lib/NativeCall/Types.pm6

```diff
@@ -76,22 +78,24 @@ our class Pointer                               is repr('CPointer') {
             )
         }
     }
-    method ^parameterize(Mu:U \p, Mu:U \t) {
-        die "A typed pointer can only hold:\n" ~
-            "  (u)int8, (u)int16, (u)int32, (u)int64, (u)long, (u)longlong, num16, num32, (s)size_t, bool, Str\n" ~
-            "  and types with representation: CArray, CPointer, CStruct, CPPStruct and CUnion" ~
-            "not: {t.^name}"
-            unless t ~~ Int|Num|Bool || t === Str|void || t.REPR eq any <CStruct CUnion CPPStruct CPointer CArray>;
-        my $w := p.^mixin: TypedPointer[t.WHAT];
-        $w.^set_name: "{p.^name}[{t.^name}]";
-        $w;
+    method ^parameterize(Mu:U $this is raw, Mu:U $type is raw) {
+        when $type ~~ any(Int, Num, Str, void)
```

Comment:
This isn't related to the problem you're fixing, but why the `when` with only one condition and then a `default`? That construct is a good bit slower than just an `if` and then an `else`.

Reply:
This used to be inside a `given` with a few different `when`s, but I forgot to change it back after removing it.